test/fixtures/gcs-fixture/src/main/java/fixture/gcs/GoogleCloudStorageHttpHandler.java

```diff
@@ -266,56 +271,62 @@ private String httpServerUrl(final HttpExchange exchange) {
-                    if (line.length() == 0 || line.equals("\r\n") || line.startsWith("--")
-                        || line.toLowerCase(Locale.ROOT).startsWith("content")) {
-                        markAndContinue = true;
-                    } else if (line.startsWith("{\"bucket\":")) {
-                        markAndContinue = true;
-                        Matcher matcher = Pattern.compile("\"name\":\"([^\"]*)\"").matcher(line);
-                        if (matcher.find()) {
-                            name = matcher.group(1);
+                            out.write(read);
+                            out.write(next);
+                            continue;
                         }
                     }
-                    if (markAndContinue) {
-                        in.mark(Integer.MAX_VALUE);
-                        continue;
+                    out.write(read);
+                } while ((read = in.read()) != -1);
+                final String bucketPrefix = "{\"bucket\":";
+                final String start = new String(out.toByteArray(), 0, Math.min(out.size(), bucketPrefix.length()), UTF_8);
+                if (start.length() == 0 || start.equals("\r\n") || start.startsWith("--")
+                    || start.toLowerCase(Locale.ROOT).startsWith("content")) {
+                    markAndContinue = true;
+                } else if (start.startsWith(bucketPrefix)) {
+                    markAndContinue = true;
+                    final String line = new String(out.toByteArray(), bucketPrefix.length(), out.size() - bucketPrefix.length(), UTF_8);
+                    Matcher matcher = NAME_PATTERN.matcher(line);
+                    if (matcher.find()) {
+                        name = matcher.group(1);
                     }
                 }
+                if (markAndContinue) {
+                    in.mark(Integer.MAX_VALUE);
+                    continue;
+                }
                 if (name != null) {
                     in.reset();
-                    try (ByteArrayOutputStream binary = new ByteArrayOutputStream()) {
-                        while ((read = in.read()) != -1) {
-                            binary.write(read);
-                        }
-                        binary.flush();
-                        byte[] tmp = binary.toByteArray();
-                        // removes the trailing end "\r\n--__END_OF_PART__--\r\n" which is 23 bytes long
-                        content = Tuple.tuple(name, new BytesArray(Arrays.copyOf(tmp, tmp.length - 23)));
+                    out.reset();
+                    while ((read = in.read()) != -1) {
+                        out.write(read);
                     }
+                    // removes the trailing end "\r\n--__END_OF_PART__--\r\n" which is 23 bytes long
+                    content = Tuple.tuple(name, new BytesArray(Arrays.copyOf(out.toByteArray(), out.size() - 23)));
+                    break;
```

Comment:
No need to continue after this one IMO, nothing can come after this part and if it does it's unexpected and should die on our EOF assertions at the end of the handler.

Reply:
Yep